in-cluster operator panic for uninitialized ProgressLog

**Bug description**

Build and deploy in-cluster operator, create IOP instance, then found that in-cluster operator panic for uninitialized mutex of ProgressLog:

```
# kubectl -n istio-operator logs -f istio-operator-5546fd668f-bphw6
2020-04-24T04:58:37.078750Z	info	ControlZ available at 127.0.0.1:9876
2020-04-24T04:58:37.378866Z	info	Registering Components.
2020-04-24T04:58:37.379277Z	info	Adding controller for IstioOperator
2020-04-24T04:58:37.379511Z	info	Controller added
2020-04-24T04:58:37.379518Z	info	Starting the Cmd.
2020-04-24T04:58:37.380096Z	info	attempting to acquire leader lease  istio-operator/istio-operator-lock...
2020-04-24T04:58:54.986160Z	info	successfully acquired lease istio-operator/istio-operator-lock
2020-04-24T04:58:57.201780Z	info	Reconciling IstioOperator
2020-04-24T04:58:57.202703Z	info	Updating IstioOperator
2020-04-24T04:58:57.202754Z	info	installer	ReadProfileYAML for profile name: demo
2020-04-24T04:58:57.202773Z	info	installer	Loading values from compiled in VFS at path profiles/demo.yaml
2020-04-24T04:58:57.203252Z	info	installer	ReadProfileYAML for profile name: default
2020-04-24T04:58:57.203272Z	info	installer	Loading values from compiled in VFS at path profiles/default.yaml
2020-04-24T04:58:57.468282Z	info	proto: tag has too few fields: "-"
2020-04-24T04:58:58.165560Z	info	Detecting third-party JWT support
2020-04-24T04:58:58.417411Z	info	Detected that your cluster does not support third party JWT authentication. Falling back to less secure first party JWT. See https://istio.io/docs/ops/best-practices/security/#configure-third-party-service-account-tokens for details.
2020-04-24T04:59:09.179854Z	info	installer	AddonComponents is waiting on dependency...
2020-04-24T04:59:09.180017Z	info	installer	Processing resources from manifest: Base for CR example-istiocontrolplane
2020-04-24T04:59:09.180227Z	info	installer	Telemetry is waiting on dependency...
2020-04-24T04:59:09.180258Z	info	installer	Pilot is waiting on dependency...
2020-04-24T04:59:09.180278Z	info	installer	Policy is waiting on dependency...
2020-04-24T04:59:09.180299Z	info	installer	IngressGateways is waiting on dependency...
2020-04-24T04:59:09.180327Z	info	installer	Cni is waiting on dependency...
2020-04-24T04:59:09.180350Z	info	installer	EgressGateways is waiting on dependency...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1572991]

goroutine 1180 [running]:
istio.io/istio/operator/pkg/util.(*ProgressLog).NewComponent(0x0, 0x20600cb, 0x4, 0x0)
	istio.io/istio/operator/pkg/util/progress.go:123 +0xa1
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).ProcessManifest(0xc000c7e280, 0xc000fd35c0, 0x1, 0x1, 0xc0004a8301, 0x0, 0x0, 0x0, 0x0, 0x0)
	istio.io/istio/operator/pkg/helmreconciler/render.go:233 +0x3bd
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).processRecursive.func1(0xc000ee96f0, 0x20600cb, 0x4, 0xc000ee96e8, 0xc000fd3680, 0xc000fd35c0, 0x1, 0x1, 0xc000c7e280)
	istio.io/istio/operator/pkg/helmreconciler/reconcile.go:174 +0x3ef
created by istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).processRecursive
	istio.io/istio/operator/pkg/helmreconciler/reconcile.go:155 +0x19f
```





**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

--- a/operator/pkg/helmreconciler/reconcile.go
+++ b/operator/pkg/helmreconciler/reconcile.go
@@ -96,7 +96,10 @@ type Options struct {
 	ProgressLog *util.ProgressLog
 }
 
-var defaultOptions = &Options{Log: clog.NewDefaultLogger()}
+var defaultOptions = &Options{
+	Log:         clog.NewDefaultLogger(),
+	ProgressLog: util.NewProgressLog(),
+}
 
 // NewHelmReconciler creates a HelmReconciler and returns a ptr to it
 func NewHelmReconciler(client client.Client, restConfig *rest.Config, iop *valuesv1alpha1.IstioOperator, opts *Options) (*HelmReconciler, error) {

